lib/tracking/tracking_manager_strategies.dart

```diff
@@ -0,0 +1,135 @@
+// Continous Strategy
+import 'dart:convert';
+import 'package:flagship/api/endpoints.dart';
+import 'package:flagship/api/service.dart';
+import 'package:flagship/cache/interface_cache.dart';
+import 'package:flagship/flagship.dart';
+import 'package:flagship/hits/activate.dart';
+import 'package:flagship/hits/batch.dart';
+import 'package:flagship/hits/hit.dart';
+import 'package:flagship/tracking/Batching/batch_manager.dart';
+import 'package:flagship/tracking/Batching/pool_queue.dart';
+import 'package:flagship/tracking/tracking_manager.dart';
+import 'package:flagship/tracking/tracking_manager_config.dart';
+import 'package:flagship/utils/constants.dart';
+import 'package:flagship/utils/logger/log_manager.dart';
+
+class TrackingManageStrategy extends TrackingManager {
+  BatchCachingStrategy strategy = BatchCachingStrategy.BATCH_CONTINUOUS_CACHING;
+
+  TrackingManageStrategy(Service service, TrackingManagerConfig configTracking,
+      IHitCacheImplementation fsCacheHit)
+      : super(service, configTracking, fsCacheHit) {
+    strategy = configTracking.batchStrategy;
+
+    // Hits pool
+    fsPool = FlagshipPoolQueue(configTracking.poolMaxSize);
+
+    // Activate pool
+    activatePool = FlagshipPoolQueue(configTracking.poolMaxSize);
+
+    // Create batch manager
+    batchManager = BatchManager(fsPool, sendBatch, configTracking, fsCacheHit);
+
+    this.delegate = batchManager;
+  }
+
+  @override
+  // Send Hit
+  Future<void> sendHit(BaseHit pHit) async {
+    Flagship.logger(Level.INFO, "Send hit with continious caching strategy");
+    if (pHit.isValid() == true) {
+      // Add to pool
+      fsPool.addTrackElement(pHit);
+
+      if (strategy == BatchCachingStrategy.BATCH_CONTINUOUS_CACHING) {
```

Comment:
I'm not familiar with the Strategy pattern, but it feels weird to have an `if` here, no ? To me, those patterns enable specific behaviors with only a single `if` at the beginning, and for every method

Reply:
😄 Initially i created 3  files for the strategies (continuous, periodic, basic_no_strategy ) , then i fusion the continuous with  periodic . 
I'll will separate them to respect the pattern . (bien vu mister )